compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -514,7 +544,7 @@ impl<'tcx> TyCtxt<'tcx> {
                 t: &Binder<'tcx, T>,
             ) -> ControlFlow<Self::BreakTy> {
                 self.outer_index.shift_in(1);
-                let result = t.as_ref().skip_binder().visit_with(self);
+                let result = t.super_visit_with(self);
```

Comment:
It reminds me that the way folding binders is done in chalk is different than rustc (there we pass the binding level IIRC).

I need to look a bit if it makes sense to skip_binder here or not.

Reply:
Passing the binding level makes sense, because every folder that deals with binders ends up doing that manually anyway. But I don't do that in this PR.

Reply:
Yeah, definitely something to do in a followup PR, not this one. :)